transport/src/main/java/io/netty/channel/SingleThreadEventLoop.java

```diff
@@ -33,6 +35,16 @@
     protected static final int DEFAULT_MAX_PENDING_TASKS = Math.max(16,
             SystemPropertyUtil.getInt("io.netty.eventLoop.maxPendingTasks", Integer.MAX_VALUE));
 
+    private final Queue<Runnable> tailTasks;
+    private final LongBiConsumer runTimeoutUpdater = new LongBiConsumer() {
+        @Override
+        public void accept(long tailTasksStartTime, long tailTasksEndTime) {
+            runTimeout -= tailTasksEndTime - tailTasksStartTime;
```

Comment:
consider clarifying the variable name `runTimeout` ... maybe something like `runAllTasksRemainingDurationNs`